Build failure: python3Packages.plumbum on aarch64-darwin

### Nixpkgs version

- Unstable (26.11)

### Steps to reproduce

```
nix shell "github:NixOS/nixpkgs/nixos-unstable-small#python3Packages.plumbum"
```

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/340763540

### Relevant log output

```console
=================================== FAILURES ===================================
_________________________ TestLocalMachine.test_pgrep __________________________

self = <test_local.TestLocalMachine object at 0x1096a40a0>

    def test_pgrep(self):
>       assert list(local.pgrep("[pP]ython"))
E       AssertionError: assert []
E        +  where [] = list(<generator object LocalMachine.pgrep at 0x109c7a5e0>)
E        +    where <generator object LocalMachine.pgrep at 0x109c7a5e0> = pgrep('[pP]ython')
E        +      where pgrep = <plumbum.machines.local.LocalMachine object at 0x108a9e980>.pgrep

self       = <test_local.TestLocalMachine object at 0x1096a40a0>

/nix/var/nix/builds/nix-9298-3887683200/source/tests/test_local.py:869: AssertionError
=========================== short test summary info ============================
SKIPPED [1] tests/test_async_remote.py:23: Skipping; marked with disabled optional tests (asyncio, skipif, ssh)
SKIPPED [1] tests/test_async_remote.py:33: Skipping; marked with disabled optional tests (asyncio, skipif, ssh)
SKIPPED [1] tests/test_async_remote.py:41: Skipping; marked with disabled optional tests (asyncio, skipif, ssh)
SKIPPED [1] tests/test_async_remote.py:50: Skipping; marked with disabled optional tests (asyncio, skipif, ssh)
SKIPPED [1] tests/test_async_remote.py:61: Skipping; marked with disabled optional tests (asyncio, skipif, ssh)
SKIPPED [1] tests/test_async_remote.py:69: Skipping; marked with disabled optional tests (asyncio, skipif, ssh)
SKIPPED [1] tests/test_async_remote.py:90: Skipping; marked with disabled optional tests (asyncio, skipif, ssh)
SKIPPED [1] tests/test_async_remote.py:104: Skipping; marked with disabled optional tests (asyncio, skipif, ssh)
SKIPPED [1] tests/test_async_remote.py:118: Skipping; marked with disabled optional tests (asyncio, skipif, ssh)
SKIPPED [1] tests/test_async_remote.py:127: Skipping; marked with disabled optional tests (asyncio, skipif, ssh)
SKIPPED [1] tests/test_async_remote.py:134: Skipping; marked with disabled optional tests (asyncio, skipif, ssh)
SKIPPED [1] tests/test_async_remote.py:141: Skipping; marked with disabled optional tests (asyncio, skipif, ssh)
SKIPPED [1] tests/test_async_remote.py:147: Skipping; marked with disabled optional tests (asyncio, skipif, ssh)
SKIPPED [1] tests/test_async_remote.py:157: Skipping; marked with disabled optional tests (asyncio, skipif, ssh)
SKIPPED [1] tests/test_async_remote.py:166: Skipping; marked with disabled optional tests (asyncio, skipif, ssh)
SKIPPED [1] tests/test_async_remote.py:175: Skipping; marked with disabled optional tests (asyncio, skipif, ssh)
SKIPPED [1] tests/test_async_remote.py:192: Skipping; marked with disabled optional tests (asyncio, skipif, ssh)
SKIPPED [1] tests/test_async_remote.py:208: Skipping; marked with disabled optional tests (asyncio, skipif, ssh)
SKIPPED [1] tests/test_async_remote.py:220: Skipping; marked with disabled optional tests (asyncio, skipif, ssh)
SKIPPED [1] tests/test_async_remote.py:231: Skipping; marked with disabled optional tests (asyncio, skipif, ssh)
SKIPPED [1] tests/test_async_remote.py:245: Skipping; marked with disabled optional tests (asyncio, skipif, ssh)
SKIPPED [1] tests/test_async_remote.py:259: Skipping; marked with disabled optional tests (asyncio, skipif, ssh)
SKIPPED [1] tests/test_examples.py:77: Skipping; marked with disabled optional tests (ssh, skipif)
SKIPPED [1] tests/test_local.py:881: Not a TTY
SKIPPED [1] tests/test_local.py:892: Not a TTY
SKIPPED [1] tests/test_remote.py:59: Skipping; marked with disabled optional tests (ssh, skipif)
SKIPPED [1] tests/test_remote.py: Skipping; marked with disabled optional tests (ssh)
SKIPPED [1] tests/test_remote.py:78: Skipping; marked with disabled optional tests (ssh, xfail)
SKIPPED [42] tests/test_remote.py: Skipping; marked with disabled optional tests (ssh, skipif)
SKIPPED [1] tests/test_remote.py:157: Skipping; marked with disabled optional tests (ssh, skipif)
SKIPPED [1] tests/test_remote.py:191: Skipping; marked with disabled optional tests (ssh, skipif, xfail)
SKIPPED [2] tests/test_remote.py:368: Skipping; marked with disabled optional tests (usefixtures, ssh, skipif, xfail)
SKIPPED [2] tests/test_remote.py:387: Skipping; marked with disabled optional tests (ssh, skipif, xfail)
SKIPPED [18] tests/test_remote.py:407: Skipping; marked with disabled optional tests (parametrize, ssh, skipif, xfail)
SKIPPED [2] tests/test_remote.py:492: Skipping; marked with disabled optional tests (parametrize, ssh, skipif)
SKIPPED [2] tests/test_remote.py:523: Skipping; marked with disabled optional tests (parametrize, ssh, skipif)
SKIPPED [1] tests/test_remote.py:605: System does not have useradd (Mac?)
SKIPPED [1] tests/test_remote.py:661: Skipping; marked with disabled optional tests (ssh, skipif, xfail)
SKIPPED [1] tests/test_sudo.py:15: Skipping; marked with disabled optional tests (sudo, skipif)
SKIPPED [1] tests/test_utils.py:13: Skipping; marked with disabled optional tests (skipif, ssh)
FAILED tests/test_local.py::TestLocalMachine::test_pgrep - AssertionError: assert []
========== 1 failed, 403 passed, 102 skipped, 2 deselected in 26.21s ===========
```

### Additional context

_No response_

### System metadata

 - system: `"aarch64-darwin"`
 - host os: `Darwin 25.4.0, macOS 26.4.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.34.8`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/store/n8gvvb3pdrw59plkwmd397ggzpl415pv-source`

### Notify maintainers

@yajo

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+label%3A%220.kind%3A+build+failure%22).
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.
- [x] I assert that I have read the [automation/AI policy](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#automationai-policy) and that this issue report complies with it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/python-modules/plumbum/default.nix
+++ b/pkgs/development/python-modules/plumbum/default.nix
@@ -2,6 +2,7 @@
   lib,
   buildPythonPackage,
   fetchFromGitHub,
+  fetchpatch,
   hatch-vcs,
   hatchling,
   mount,
@@ -28,6 +29,14 @@ buildPythonPackage rec {
     tag = "v${version}";
     hash = "sha256-i99HpT/QuF9JwX92IwwOqpEVUc/1k39E7N9v9TZ4Qvg=";
   };
+
+  patches = [
+    # Fix parsing processes with an empty STAT field on Darwin
+    (fetchpatch {
+      url = "https://github.com/tomerfiliba/plumbum/commit/217a22f361ea590c77da5f73d1387fe8a57fd742.patch";
+      hash = "sha256-Cp01aeTkJS5wy6ZZs95suzKV/HKziy1YIQtk4jYT2Gg=";
+    })
+  ];
 
   build-system = [
     hatchling

